docs/rackhd/graphs.rst

```diff
@@ -31,6 +31,287 @@ Graphs are defined via a JSON definition that conform to this schema:
+    }
+
+As is, this task definition specifies default values of "value 1" and "value 2" for its respective options. 
+In the graph definition, these values can be changed to have new defaults by adding a key to the ``Graph.options`` object
+that matches the ``label`` string given to the task object ("example-options-task" in this case):
+
+::
+
+    {
+        "injectableName": "Graph.Example.Options",
+        "friendlyName": "Override options for a task",
+        "options": {
+            "example-options-task": {
+                "option1": "overridden value 1",
+                "option2": "overridden value 2"
+            }
+        },
+        "tasks": [
+            {
+                "label": "example-options-task",
+                "injectableName": "Task.Example.Options"
+            }
+        ]
+    }
+
+    // Task.Example.Options will be run as this
+    {
+        "injectableName": "Task.Example.Options",
+        "friendlyName": "Task with basic options",
+        "implementsTask": "Task.Base.Example",
+        "options": {
+            "option1": "overridden value 1",
+            "option2": "overridden value 2"
+        },
+        "properties": {}
+    }
+
+Sometimes, it is necessary to be able to propagate the same values to multiple tasks, but it can be a chore
+to make a separate options object for each task label. In this case, there is a special field
+used in the ``Graph.options`` object called ``defaults``.
+When ``defaults`` is set, the graph will iterate through each key in the object and override that value for every task definition
+that also has that key in its respective options object. In the above example, the ``Task.Example.Options`` definition will be changed with new
+values for ``option1`` and ``option2``, but not for ``option3``, since ``option3`` does not exist in the options object for that task definition:
+
+:: 
+    {
+        "injectableName": "Graph.Example.Defaults",
+        "friendlyName": "Override options with defaults",
+        "options": {
+            "defaults": {
+                "option1": "overridden value 1",
+                "option2": "overridden value 2",
+                "option3": "this will not get set"
+            }
+        },
+        "tasks": [
+            {
+                "label": "example-options-task",
+                "injectableName": "Task.Example.Options"
+            }
+        ]
+    }
+
+    // Task.Example.Options will be run as this
+    {
+        "injectableName": "Task.Example.Options",
+        "friendlyName": "Task with basic options",
+        "implementsTask": "Task.Base.Example",
+        "options": {
+            "option1": "overridden value 1",
+            "option2": "overridden value 2"
+        },
+        "properties": {}
+    }
+
+.. _Graph Username Example:
+
+The ``defaults`` object can be used to share values across every task definition that includes them, 
+as in this example workflow that validates and sets a username.
```

Comment:
The grammar rules here are a little tricky, although "username" starts with a vowel, you use "a" because the "u" in "username" is pronounced with a "y" sound. In these cases, you use "a" instead of "an" so that it's easier/more pleasant to speak (see the bottom of this page: https://owl.english.purdue.edu/owl/resource/591/01/).